diff --git a/src/bin/zjstatus.rs b/src/bin/zjstatus.rs
--- a/src/bin/zjstatus.rs
+++ b/src/bin/zjstatus.rs
@@ -259,7 +259,7 @@ impl State {
 
                 should_render = true;
             }
-            Event::PaneUpdate(pane_info) => {
+            Event::PaneUpdate(mut pane_info) => {
                 tracing::Span::current().record("event_type", "Event::PaneUpdate");
                 tracing::debug!(pane_count = ?pane_info.panes.len());
 
@@ -277,6 +277,14 @@ impl State {
                     false,
                 );
 
+                for panes in pane_info.panes.values_mut() {
+                    for pane in panes {
+                        if pane.is_focused && !pane.is_plugin {
+                            pane.title = "Pane #1".to_owned();
+                        }
+                    }
+                }
+
                 self.state.panes = pane_info;
                 self.state.cache_mask = UpdateEventMask::Tab as u8;
 
